### ISSUE TITLE
New possibility to smelt sticks and bamboo into charcoal pieces?

### ISSUE THREAD

**Issue by gniftygnome** [Apis035/charcoal-plus#1]
Smelting a log into charcoal consumes 1 fuel and converts 1.5 fuel to 8 fuel.  The efficiency is 14:3.  I think there may be a good opportunity here to do something similar with sticks and/or bamboo at a lower efficiency.  However, because the output is only 1 fuel, burn time would need to be tweaked to get the desired efficiency.  What do you think about the following two recipes?

sticks:  1 stick to 1 charcoal piece, cook time 1 second, efficiency 9:5
bamboo:  1 bamboo to 1 charcoal piece, cook time 7.5 seconds, efficiency 1:1

Thus compared to the nearly 3 times better efficiency of charcoal vs. logs, you would get a nearly 2 times better efficiency for smelting sticks and no efficiency gain (1 times better) for smelting bamboo.  However, smelting bamboo would still be worth it because of the XP and the improved storage efficiency.  I think these additional fuel pathways would be really useful for technical play, since they don't require player intervention (crafting).  Bamboo in particular as a furnace array fuel is quite challenging to move around at the rate required to maintain furnace activity.

Proposed recipes:

Sticks:
```json
    {
      "type": "minecraft:smelting",
      "ingredient": {
        "item": "minecraft:stick"
      },
      "result": "charcoalplus:charcoal_piece",
      "experience": 0.02,
      "cookingtime": 20
    }
```

Bamboo:
```json
    {
      "type": "minecraft:smelting",
      "ingredient": {
        "item": "minecraft:bamboo"
      },
      "result": "charcoalplus:charcoal_piece",
      "experience": 0.1,
      "cookingtime": 150
    }
```

**Comment by Apis035** [comment]
Sounds like a good addition 👍

But I think your suggestion needs more tweaking, ignoring the source item fuel efficiency (log 1.5, stick 0.5, bamboo 0.25):
- I'd like to set the `cookingtime` for stick to 25 to make the smelting time and efficiency 1:1 with log → charcoal (200). So smelting 8 stick is equal to smelting 1 log (1 log = 4 plank = 8 stick)
- Bamboo is a 0.25 fuel, you need 4 bamboo to use it as a fuel to smelt 1 item. To make 4 bamboo equal to 1 fuel, `cookingtime` should be set to 200 to make it balanced with the resulting charcoal.

For comparison, charcoal piece has 200 tick `cookingtime` (1 fuel)

**Comment by gniftygnome** [comment]
> ignoring the source item fuel efficiency (log 1.5, stick 0.5, bamboo 0.25)

I don't agree you can entirely ignore the input.  The amount of fuel consumed in the conversion is the biggest factor in the value of the output.  It's not a big deal with the sticks but with the bamboo what you are suggesting is to actually lose fuel value from the conversion which does not make much sense to me.  Charcoal is made because of its much greater fuel efficiency.  Here are the conversion efficiencies from the longer burn times you suggest (and log -> charcoal for comparison):

```
logs: 1 log to 1 charcoal, cook time 10 seconds,
        conversion efficiency: (8 output - 1 input) / (1.5 source item fuel) = 14/3
sticks: 1 stick to 1 charcoal piece, cook time 1.25 seconds,
        conversion efficiency: (1 output - 0.125 input) / (0.5 source item fuel) = 7/4
bamboo: 1 bamboo to 1 charcoal piece, cook time 10 seconds,
        conversion efficiency: (1 output - 1 input) / (0.25 source item fuel) = 0
```

For the sticks what you can see is although it is quite a good deal to convert a log into charcoal (almost five times better than burning the log), it is very much less good to convert a stick into a charcoal piece (less than twice as good as merely burning the stick).  This is particularly ironic since in the real world historically charcoal was most often made from collected sticks.  It is still an OK deal but even under the values I suggested it will never make sense to convert a log to sticks to smelt them.  Also, ensuring this does not make much sense even for XP is why I set the XP of the stick recipe so low.

For the bamboo, under my proposed recipe you would get exactly the same amount of fuel value out as you put in.  You put in 4 bamboo (1 in the top and 3 in the fuel slot) and you get 1 charcoal piece with a burn time the same as 4 bamboo.  In my opinion this is the bare minimum return on investment that makes it worth cooking the bamboo.  To sweeten the deal a bit I also set the XP in the recipe a bit higher.  For contrast, with a 1 fuel burn time (200 ticks), you would take the charcoal piece out of the bottom of the furnace and put it right into the fuel slot to cook the next bamboo, completely consuming it in order to produce another charcoal piece and repeat the cycle while gaining nothing but a little XP.

**Comment by Apis035** [comment]
I was messed up with the calculation.

Bamboo is okay with your proposed recipe.
But I still don't quite agree with the extra efficiency coming from smelting stick. With 20 `cookingtime` you can get 10 charcoal pieces using 1 fuel against getting 8 charcoal pieces from smelting 1 log using 1 fuel. I was trying to keep this mod close to vanilla as possible.

**Comment by gniftygnome** [comment]
Yes, I had noticed the cook rate was a little faster than that of logs.  I was trying to balance it so the fuel efficiency of the stick recipe was sort of in between the other two.  Also it *is* a lot faster to make charcoal out of sticks than logs...  However although I personally think the recipe I proposed is the best balance, in terms of game play there is little difference between the two.  Actually at your higher XP rate, even with a slower cook rate the stick recipe may be a better deal for technical players since the XP is so much higher than what you get from cooking it as a log.  So I'm happy enough if you're willing to adjust the bamboo recipe.  :)
